x-pack/plugins/osquery/public/results/results_table.tsx

```diff
@@ -88,30 +98,59 @@ const ResultsTableComponent: React.FC<ResultsTableComponentProps> = ({ actionId
   );
 
   useEffect(() => {
-    const newColumns: EuiDataGridColumn[] = keys(results[0]?.fields)
+    // @ts-expect-error update types
+    if (!allResultsData?.results) {
+      return;
+    }
+    // @ts-expect-error update types
+    const newColumns = keys(allResultsData?.results[0]?.fields)
       .sort()
-      .map((fieldName) => ({
-        id: fieldName,
-        displayAsText: fieldName.split('.')[1],
-        defaultSortDirection: 'asc',
-      }));
+      .reduce((acc, fieldName) => {
+        if (fieldName === 'agent.name') {
+          return [
```

Comment:
similar comments here as the other usage of `reduce`, this could be done with a map and filter, or swapping the spread copying with modification of the `acc` value

Reply:
you're right, it's going to be refactored when we have a decision about columns in the results table